internal/analytics/tilt_analytics.go

```diff
@@ -0,0 +1,71 @@
+package analytics
+
+import (
+	"time"
+
+	"github.com/windmilleng/wmclient/pkg/analytics"
+)
+
+// An Analytics that:
+// 1. Has `IncrIfUnopted` to report anonymous metrics only for users who have not opted in/out (or the choice that they
+//    did opt in/out).
+// 2. Ignores all other calls from users who have not opted in.
+// 3. Allows opting in/out at runtime.
+type TiltAnalytics struct {
+	opt        analytics.Opt
+	persistOpt optPersister
+	a          analytics.Analytics
+}
+
+type optPersister func(analytics.Opt) error
+
+func NewTiltAnalytics(opt analytics.Opt, setOpt optPersister, analytics analytics.Analytics) *TiltAnalytics {
+	return &TiltAnalytics{opt, setOpt, analytics}
+}
+
+func (ta *TiltAnalytics) Count(name string, tags map[string]string, n int) {
+	if ta.opt == analytics.OptIn {
+		ta.a.Count(name, tags, n)
+	}
+}
+
+func (ta *TiltAnalytics) Incr(name string, tags map[string]string) {
+	if ta.opt == analytics.OptIn {
+		ta.a.Incr(name, tags)
+	}
+}
+
+func (ta *TiltAnalytics) IncrIfUnopted(name string, tags map[string]string) {
```

Comment:
arguably people can do even less weird stuff w this func if it doesn't accept tags. do we even want to let ppl pass tags here?